### ISSUE TITLE
pyload not reachable using ssl with valid certificate

### ISSUE THREAD

**Issue by segelkma** [Flexget/Flexget#718]
I am getting the error pyload not reachable using SSL in pyloads web interface with my own ssl certificate. The certificate is a free one from www.startssl.com
The certificate is set up correctly (can access via browser with certificate being valid). But in flexget i get the error:
    CRITICAL pyload        grab_test       pyLoad not reachable

As api url I used the domain adress and the local adress (localhost, just changed from http as I used before to https). I understand why the local adress fails, but not the domain adress. I also tried setting verify_ssl_certificates to no and then it works for both methods. So it seems that flexget doesnt accept the certificate neither way, but it should at least when I set up the domain url as api url.

**Comment by gazpachoking** [comment]
First thing to try with any ssl issues is to upgrade python to >= 2.7.9, or if it isn't available on your platform, `pip install requests[security]`.

**Comment by segelkma** [comment]
Python version running is/was 2.7.10, sorry, forgot to mention it. No virtualenv is used.

**Comment by qvazzler** [comment]
@segelkma Are you having issues still? There have been SSL updates in more than one place recently so I'd like to make sure.

**Comment by segelkma** [comment]
@qvazzler Sorry but I cannot test it. Switched to JDownloader because pyload was running so poorly and right now I have no reason at all to go back to pyload.

**Comment by qvazzler** [comment]
@segelkma no problem.

(Closing until further notice.)

**Comment by segelkma** [comment]
Hello there, I am back. Recently I started using pyload again and today I wanted to test if the API works using ssl. It does not. Difference to last time: flexget runs under python3.5 (pyload still 2.7).
Webinterface of pyload is reachable via https. What I don't understand: Is there an issue dealing with it from the site of flexget or is the error made up by pyload...
http runs fine, I'm doing nothing but enabling https in pyload and changing the config url of pyload from http to https.
> 2016-11-23 17:30 DEBUG    utils.requests test            Fetching URL https://mydomain.de:8000/api/login with args () and kwargs {'json': None, 'data': {u'username': u'IAmAUser', u'password': u'IAmAPassword'}, u'timeout': 30}
> 2016-11-23 17:30 CRITICAL pyload        test            pyLoad not reachable

**Comment by segelkma** [comment]
Alright, just found out that pyload is reachable but my certificate is not accepted. When I add `verify_ssl_certificates: no` then it works...
Last year I used a free certificate by StartCom, now it's coming from Lets Encrypt. Why are those certificates not accepted by default?

**Comment by liiight** [comment]
hmm, maybe update cryptography? 
`pip install cryptography -U`

**Comment by segelkma** [comment]
Did not do the magic. So it is an issue with one of the python plugins, but not directly with flexget right?

**Comment by liiight** [comment]
probably. try `pip install requests[security] -U`

**Comment by segelkma** [comment]
No change

**Comment by segelkma** [comment]
Did some more testing today and this might help. Looks like the certificate does not get rejected in general. I created my own xml file and stored it on my webhost, using it as source with rss plugin. And it works, the xml is being parsed while pyload in the same tasks fails. It is the same certificate.

I also tested if the pyload api is accessable via https with postman chrome addon and it gave me the same results for http and https.
